api/envoy/extensions/filters/http/cache/v3/cache.proto

```diff
@@ -52,6 +52,14 @@ message CacheConfig {
 
   // Config specific to the cache storage implementation. Required unless ``disabled``
   // is true.
+  //
+  // Note that when the cache is enabled, cacheable requests are only sent through
+  // filters in the ``upstream_filters`` chain and *not* through any filters in the
+  // regular filter chain that are further upstream than the cache filter, while
+  // non-cacheable requests still go through the listener filter chain. It is
+  // recommended that only the router filter should be further upstream in the
+  // listener filter chain than the cache filter.
```

Comment:
Not sure I fully understand.
Do `upstream_filters` refer to [upstream_http_filters](https://github.com/envoyproxy/envoy/blob/aa195eb01d1211aa9e082f2d742d75395975edc0/api/envoy/extensions/filters/http/router/v3/router.proto#L136)?

If so, it is unclear to me why a request skips the downstream-filters and only goes through the upstream filters.

Just a suggestion to make this a bit more understandable: it would be nice to have a diagram that depicts which filters are invoked and which are not (similar to https://www.envoyproxy.io/docs/envoy/latest/intro/life_of_a_request#http-filter-chain-processing where there are arrows that skip some filters.

Reply:
Yes, `upstream_http_filters`. It's not that it skips the downstream-filters, it's that it skips the filter chain in between the router filter and the cache filter, because that's just what the `AsyncHttpRequest` interface does - it's not an intentional behavior, it's an unfortunate concession to how things work internally.

Moved this into the cache rst document because I don't think we can do images links in generated protodoc, and it's probably the more appropriate place for it anyway, and clarified and linked and added an svg.

Reply:
(Also updated an outdated statement in that doc.)